### PR TITLE
feat: Integrate WDS Menu Button Widget

### LINKED ISSUE #28864: Integrate WDS Menu component

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description
Adds WDS Menu Button Widget.

#### PR fixes following issue(s)
Fixes #28864
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `Menu` component with customizable color options for items.
  - Added dynamic styling for menu items based on color attributes.
  - Implemented a new `WDSMenuButtonWidget` with configurable properties and styles.
  - Expanded the property pane with new configurations for menu items and buttons.

- **Enhancements**
  - Improved menu placement options and handling of colors within the Menu component.
  - Enhanced the `MenuButtonWidget` with additional properties for better customization.

- **Bug Fixes**
  - Fixed incorrect component naming for `MenuList` to ensure consistency.

- **Documentation**
  - Updated Menu component stories to reflect new color handling features.

- **Refactor**
  - Streamlined export statements for Menu components to simplify imports.
  - Organized property pane configurations into separate modules for clarity.

- **Style**
  - Added new styles for the search bar and menu items for a more polished UI.

- **Tests**
  - No visible changes to end-users.

- **Chores**
  - No visible changes to end-users.

- **Revert**
  - No visible changes to end-users.

### CODE REVIEW

File: app/client/src/widgets/wds/WDSMenuButtonWidget/config/propertyPaneConfig/childPanels/configureMenuItemsConfig.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+          evaluatedDependencies: ["sourceData"],
+        },
+      ],
+    },
+    {
+      sectionName: "Color",
+      children: [
+        {
+          propertyName: "textColor",
+          label: "Text Color",
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/WDSMenuButtonWidget/config/propertyPaneConfig/childPanels/configureMenuItemsConfig.ts:160]: @ichik  So let's get rid of this? 
Снимок экрана 2023-12-04 в 16 11 00

**Review by KelvinOm — DISMISSED** [review]

File: app/client/src/widgets/wds/WDSMenuButtonWidget/config/propertyPaneConfig/childPanels/configureMenuItemsConfig.ts (hunk 1 above)
Comment (KelvinOm) [reply app/client/src/widgets/wds/WDSMenuButtonWidget/config/propertyPaneConfig/childPanels/configureMenuItemsConfig.ts:160]: @ichik I'm talking about the property pane and about removing it altogether this option from it for now.

**Review by KelvinOm — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Pull request merged**